### ISSUE TITLE
Fix false positives for variables in font-family-no-missing-generic-family-keyword

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#4765]
> Clearly describe the bug

There are false positives in `font-family-no-missing-generic-family-keyword` if I use a variable _and a font family_.

[Demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIA6A7AhgAmKjGBmEKALgLR5oC2AlgDYCe8GA5AcUwDQYBuaATgBQkyhIgEoA3KgC+IdiADOdYmgAeCcPPmzwhPFQDm67ClxIQvAK40Y8s42O5HZ1qXLV6JFBBLVNVFPok+jAoMLxUYGSUtHQkANYwdADuELxQdhgA2kSWMJw4jo7AGGYGXrwwAGIildE0VDYZmWY8AkIuomYAuhhSBY5dBX0oMlJAA).

See https://github.com/stylelint/stylelint/issues/3848#issuecomment-617288304.

> Which rule, if any, is the bug related to?

`font-family-no-missing-generic-family-keyword`

> What code is needed to reproduce the bug?

```css
a {
  font-family: 'font', var(--font);
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
     "font-family-no-missing-generic-family-keyword": [true, 
       { "ignoreFontFamilies": ["var(--font)"] }
     ]
  }
}
```

> Which version of stylelint are you using?

13.3.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Demo

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

No warnings to be flagged because the rule should ignore values that _contain_ variables.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
3:16 error Unexpected missing generic font family (font-family-no-missing-generic-family-keyword)
```

**Comment by jeddy3** [comment]
I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by lensbart** [comment]
The same issue still persists with Sass variables (e.g. `font-family: 'Comic Sans', $fontFamilyUI;`)

**Comment by ybiquitous** [comment]
@lensbart Thanks for the report. We can reproduce it on the [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYKBmEB2AXAtBkgLYCWANgJ7woDkAwhGWCgMpLYDOtANCgG5IATgAp8hHASJkq+AK6kAlAG4UAXwA62VOix5CJCtTqNmbDtz4ASPbgBihqgFUAkqs3YQPEJ0p4kAB4I4JycXuA4GKQA5sHAWigoGiBCcuQwnMk08diJicm2BjKU+NgQ+GShpNjR+NEw2DBCpGBFRvgA1jCUAO4QQlBZKLipMAnqWmogakA).

**Comment by ybiquitous** [comment]
I've opened a new PR #5240!
